### DISCUSSION TITLE
wt32-sc01-3.5

### DISCUSSION THREAD

**Discussion by xHirscHx** [HASwitchPlate/openHASP#102] [General]
Hi all hope someone can help me. I have 5 HASP (originals)all setup running  2 on new firmware . I bought a wt32_sc01- 3.5 and saw this form. Im no coder just a copy paste guy . 2 questions.
#1  Iv got The wt32 up and running . I  just  copy the demo page code to a text document then rename it and upload it,is that right or an easier way of doing (demo page is for 2.4 they don't fit). anybody have any 3.5 demos.
#2 is this like old HASP and uses packages or new and uses blueprints.
This looks fun I like the fact i can use gpios if i can ever figure it out. I want to use the base of a sonoff touch for its relays and power an place this over it so i don't have to cut anymore holes in my walls(LOL).

**Comment by fvanroie** [comment]
Hello,

- 1 You can also send the pages.jsonl contents to the `hasp/platename/command/jsonl` mqtt topic. There is a limit up to 2Kb though, so large files need to be split up into multiple messages. For testing you can also type `json {"obj":"btn","id":1}` into the serial console and see the result immediately. We are working on a way to deploy the [pages as blueprints](https://github.com/blakadder/hasplvgl-hablueprints) too.
- 2 This project has diverged a lot from the HASP codebase over the past 15 months, so you should consider it as a separate device. Some things will seem familiar, but this project is not like the HASP 0.4.0 or HASPone firmware. The HA integration part is still ongoing.

It's a cool idea to use a Sonoff base and it might actually work 😃. I'm currently testing the firmware on a [Lanbon L8](https://blakadder.com/lanbon-L8-custom-firmware/), which is like a Sonoff touch with integrated LCD. The WT32-SC01 has plenty of GPIO's so that shouldn't be a problem.

There is a [HACS custom component](https://github.com/dgomes/hasp-lvgl-custom-component) in the works for HA too that should simplify integration. But you need to have the hasp-lvgl 0.4.0-dev version installed.

**Comment by xHirscHx** [comment]
Can i use the update to load the 0.4.0-dev or do i have to start over.

**Reply by fvanroie** [reply]
You need currently to upload 0.4.0-dev via Platformio atm, because the partitionscheme has changed.

**Reply by xHirscHx** [reply]
anybody out there using a 3.5  screen and what to share the dimensions for the buttons to change in the pages.jsonl to fit the screen .

**Comment by htvekov** [comment]
Hi' @xHirscHx

I'm using the wt32-sc01 devices and has made a few revisions to page designs to fit the wt32-sc01 larger screen resolution.
Have not really spend much time on the designs yet, as I've been focusing on the fw capabilities and automations.
The resolution is 1,33 larger in widht and 1,5 larger in height (in portrait mode), so it not that difficult to calculate.

You'll probably also need to load a larger custom font (size 36 or 40) which roughly matches standard font size 28 on the smaller 230x320 screen.
@fvanroie has just recently fixed the zi font issues (well, most of them) and it's working again.
Finally I would definitely suggest that you use @dgomes cc for HA. It's, as hasp-lvgl itself, under rapid development. All 'basic' tasks is already handled by this cc (simple devices state tracking, use of template, page switch, dimming, screen on/off etc.) If you need something more fancy, you'll have to do the automations yourself atm. 😉 

Below my preliminary page zero design and one simple page with five buttons and a top header:

```
{"page":0,"comment":"---------- All Pages ----------"}
{"obj":"btn","id":1,"page":0,"x":000,"y":400,"w":104,"h":80,"bg_color":"#000000","border_color":"#C7BAA7","border_width":2,"radius":10,"radius1":10,"radius2":10,"txt":"\uf060","text_color":"#978B7D","text_font":28}
{"obj":"btn","id":2,"page":0,"x":109,"y":400,"w":102,"h":80,"bg_color":"#000000","border_color":"#C7BAA7","border_width":2,"radius":10,"radius1":10,"radius2":10,"txt":"\uf0C9","text_color":"#978B7D","text_font":28}
{"obj":"btn","id":3,"page":0,"x":216,"y":400,"w":104,"h":80,"bg_color":"#000000","border_color":"#C7BAA7","border_width":2,"radius":10,"radius1":10,"radius2":10,"txt":"\uf061","text_color":"#978B7D","text_font":28}

{"page":3,"comment":"---------- Page 3 ----------"}
{"obj":"btn","id":8,"x":0,"y":000,"w":320,"h":45,"txt":"OFFICE","text_font":28,"bg_color":"#2C3E50","text_color":"#FFFFFF","border_side":0,"radius":0}
{"obj":"btn","id":4,"x":0,"y":052,"w":320,"h":80,"bg_color":"#000000","border_color":"#FFAC00","border_width":2,"radius":10,"radius1":10,"radius2":10,"toggle":true,"txt":"Lights On","value_ofs_x":-85,"value_font":28,"value_str":"\uf205","value_color":"#B6B6B6","text_color":"#B6B6B6","text_font":28}
{"obj":"btn","id":5,"x":0,"y":139,"w":320,"h":80,"bg_color":"#000000","border_color":"#FFAC00","border_width":2,"radius":10,"radius1":10,"radius2":10,"toggle":true,"txt":"Office desk","value_ofs_x":-85,"value_font":28,"value_str":"\uf185","value_color":"#B6B6B6","text_color":"#B6B6B6","text_font":28}
{"obj":"btn","id":6,"x":0,"y":226,"w":320,"h":80,"bg_color":"#000000","border_color":"#FFAC00","border_width":2,"radius":10,"radius1":10,"radius2":10,"toggle":true,"txt":"Night","value_ofs_x":-85,"value_font":28,"value_str":"\uf186","value_color":"#B6B6B6","text_color":"#B6B6B6","text_font":28}
{"obj":"btn","id":7,"x":0,"y":313,"w":320,"h":80,"bg_color":"#000000","border_color":"#FFAC00","border_width":2,"radius":10,"radius1":10,"radius2":10,"txt":"Not used","value_ofs_x":-85,"value_font":28,"value_str":"\uf204","value_color":"#B6B6B6","text_color":"#B6B6B6","text_font":28}
```

**Reply by nagyrobi** [reply]
How to use zi fonts?
Also how ro create custom zi fonts, for example with MDI icons?

**Reply by nagyrobi** [reply]
I looked at https://github.com/hagronnestad/nextion-font-editor but it looks like only supports latin codepage.
MDI icons are on different plane offsets in the font file, but I wasn't able to find a place in the app to select those.

**Reply by xHirscHx** [reply]
Great thx. Its a good look fits screen nicely. Iv already install the custom component still playing with it. If you do any more screens would love to see them. I have a 2 year old and i get about an hour a night for me time so hard to get things done.
thx again.

**Reply by xHirscHx** [reply]
Was also wondering if you figured out or plain to incorporate the gpios of the  wt32-sc01 board , to trigger relays.

**Reply by htvekov** [reply]
At the moment, no. Almost all my wall switches are already fitted with Tasmotized Shelly One's / Sonoff Mini's that via rules commands controls my Hue/Z2M lights. I only need to be able to control relays directly if HA / MQTT broker for some reason is down. So when hasp-lvgl at some point gets a stable local setup running, I'll implement a GPIO setup as well to control the Sonos Mini's relays directly.

For now I've only implemented a PIR sensor using a GPIO pin to wakeup device / keep device awake when presence is detected.

**Reply by xHirscHx** [reply]
Im lost. using your page config im able to set automations to toggle  lights on and off but i can't figure out how to sync it with the light on the button on the plate. also custom component i can't figure out what the difference is between these----
openhasp:  (Required) Your platform identifier--------------------------------------no clue                                            
plate_my_room: (Required) Your plate identifier ------------------------------------no clue                                           
topic:: (string) (Required) The MQTT topic your plate is configured with.----------got this on

**Reply by nagyrobi** [reply]
See [documentation](https://haswitchplate.github.io/openHASP-docs/#custom-component/configuration/) scroll down to `Configuration tips`. Also see the [basics](https://haswitchplate.github.io/openHASP-docs/#custom-component/sampl_conf_basics/)

**Comment by htvekov** [comment]
Just load a zi font (via plates webpage -> hasp design) and the correcsponding sized fonticons font.
(There's a link to zi packs somewhere in the hasp-lvgl documentation.)
Mark the loaded zi font as default font on the hasp design page as well.

Use `p[x]b[y].text_font=0` to select loaded font for object(s)

Zi fonts still somewhat 'flaky' I believe. But I've loaded one zi font and used it without issue.
It could crash the device though and make it unreachable. You'll have to serial flash fra scratch.

No idea how to create these font packages.

**Reply by nagyrobi** [reply]
oh, so the text_font has to be 0 ??

I tried but after setting to default, device bricked. Had to reflash.

**Reply by nagyrobi** [reply]
FontAwesome has too few icons + they don't look as nice as MDI. And with MDI you could make an interface looking very similar with HA's Lovelace UI.
So it would be great to have a way to import a collection of MDI icons in .zi format.

**Reply by htvekov** [reply]
Yep. text_font=0 (enables the default loaded font)
Haven't had the time yet to check what icons actually are included in the fontawesome files.

I've experienced crashes too.
Implementation is still somewhat not quite ready and stable.
@fvanroie wrote that there was still some work to be done on this.

**Comment by RossiRPI** [comment]
The wt32-sc01 has a footprint for a vcnl4030x01 (U1) which is a Proximity and Ambient Light Sensor which supports gesture recognition. Can we add support for it to the main build of openHASP?

**Reply by fvanroie** [reply]
You would have to create a custom build for this and then add the appropriate libraries to read the sensor.
As this footprint is not populated by default, I don't think there will be a lot of users that will add the IC.
Do you have a sketch that already works with this sensor?

**Reply by RossiRPI** [reply]
I haven't had a chance to look into it. I have ordered a few vcnl4030x01 which I'll populate and test out.

**Reply by g7wbd** [reply]
thankyou  how did they work out ?
